custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,545 @@
+    # aliases found in xml configs
+    "comfort": ("setpoint_heating",),
+    "heat mode": ("setpoint_heating",),
+    "heat (default)": ("setpoint_heating",),
+    "dry floor": ("setpoint_dry_air",),
+    "heat eco": ("setpoint_eco_heating",),
+    "energy saving": ("setpoint_eco_heating",),
+    "energy heat": ("setpoint_eco_heating",),
+    "vacation": ("setpoint_away_heating", "setpoint_away_cooling"),
+    # for tests
+    "heat_cool": ("setpoint_heating", "setpoint_cooling"),
+}
+
+HVAC_CURRENT_MAPPINGS = {
+    "idle": CURRENT_HVAC_IDLE,
+    "heat": CURRENT_HVAC_HEAT,
+    "pending heat": CURRENT_HVAC_IDLE,
+    "heating": CURRENT_HVAC_HEAT,
+    "cool": CURRENT_HVAC_COOL,
+    "pending cool": CURRENT_HVAC_IDLE,
+    "cooling": CURRENT_HVAC_COOL,
+    "fan only": CURRENT_HVAC_FAN,
+    "vent / economiser": CURRENT_HVAC_FAN,
+    "off": CURRENT_HVAC_OFF,
+}
+
+PRESET_MAPPINGS = {
+    "away": PRESET_AWAY,
+    "full power": PRESET_BOOST,
+    "manufacturer specific": PRESET_MANUFACTURER_SPECIFIC,
+}
+
+DEFAULT_HVAC_MODES = [
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_DRY,
+    HVAC_MODE_OFF,
+    HVAC_MODE_AUTO,
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Z-Wave Cover from Config Entry."""
+
+    @callback
+    def async_add_climate(values):
+        """Add Z-Wave Climate."""
+        climate = None
+        if values.primary.command_class == CommandClass.THERMOSTAT_SETPOINT:
+            climate = ZWaveClimateSingleSetpoint(values)
+        elif values.primary.command_class == CommandClass.THERMOSTAT_MODE:
+            climate = ZWaveClimateMultipleSetpoint(values)
+        async_add_entities([climate])
+
+    async_dispatcher_connect(hass, "zwave_new_climate", async_add_climate)
+    await hass.data[DOMAIN][config_entry.entry_id]["mark_platform_loaded"]("climate")
+
+
+class ZWaveClimateBase(ZWaveDeviceEntity, ClimateDevice):
+    """Representation of a Z-Wave Climate device."""
+
+    def __init__(self, values):
+        """Initialize the Z-Wave climate device."""
+        ZWaveDeviceEntity.__init__(self, values)
+        self._target_temperature = None
+        self._target_temperature_range = (None, None)
+        self._current_temperature = None
+        self._hvac_action = None
+        self._hvac_list = None  # [zwave_mode]
+        self._hvac_mapping = None  # {ha_mode:zwave_mode}
+        self._hvac_mode = None  # ha_mode
+        self._aux_heat = None
+        self._default_hvac_mode = None  # ha_mode
+        self._preset_mapping = None  # {ha_mode:zwave_mode}
+        self._preset_list = None  # [zwave_mode]
+        self._preset_mode = None  # ha_mode if exists, else zwave_mode
+        self._current_fan_mode = None
+        self._fan_modes = None
+        self._fan_action = None
+        self._current_swing_mode = None
+        self._swing_modes = None
+        self._unit = None
+        self._zxt_120 = None
```

Comment:
Is this workaround still necessary or should that live in openzwave?

Reply:
please leave all workarounds out because most of it has been addressed upstream. When we get issues reported we will look into that and see if it can be fixed upstream, if not a workaround in this code needs to be done.

In short: Do not blindly copy workarounds from the old implementation because the new implementation uses OZW 1.6 which addresses most of those shortcomings,